examples/pytorch/text-classification/Fine-Tune-Llama3-LLM.ipynb

```diff
@@ -0,0 +1,394 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "270ce448",
+   "metadata": {},
+   "source": [
+    "# Fine-Tune & Serve Llama3 with Kubeflow PytorchJob in a Kubeflow Pipeline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "73619a20",
+   "metadata": {},
+   "source": [
+    "This Notebook will do the following:\n",
+    "1. Fine-tune meta-llama/Llama-3.1-8B-Instruct model on KubeCon, India 2024 dataset using distributed training with [Kubeflow PytorchJob](https://www.kubeflow.org/docs/components/training/overview/).\n",
+    "2. Serve the fine-tuned model using Kserve.\n",
+    " \n",
+    "We are using [Kubeflow Pipelines](https://www.kubeflow.org/docs/components/pipelines/) to run this end-to-end LLM pipeline.\n",
+    "\n",
+    "\n",
+    "Llama3 model: https://huggingface.co/meta-llama/Llama-3.1-8B-Instruct\n",
+    "\n",
+    "KubeCon, India 2024 dataset: https://huggingface.co/datasets/aishwaryayyy/events_data\n",
+    "\n",
+    "This Notebook requires:\n",
+    "1. 1 GPU on your Kubernetes cluster for fine-tuning and later serving the fine-tuned model\n",
+    "2. 1 GPU on your Notebook node to load the fine-tuned model by merging PEFT weights.\n",
+    "\n",
+    "We need to install Kubeflow Pipeline packages and import the dependencies."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8e002b46-a18d-4805-ab57-9be5ed7a07eb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pip install kfp kfp-kubernetes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "af5e73b2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from typing import List\n",
+    "from kfp import client\n",
+    "from kfp import dsl\n",
+    "from kfp.dsl import Dataset\n",
+    "from kfp.dsl import Input\n",
+    "from kfp.dsl import Model\n",
+    "from kfp.dsl import Output"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1199e8b6",
+   "metadata": {},
+   "source": [
+    "## Fine-Tune Llama3 model with KubeCon dataset\n",
+    "\n",
+    "In this component, use TrainingClient() to create PyTorchJob which will fine-tune Llama3 model on 1 worker with 1 GPU.\n",
+    "\n",
+    "Specify the required packages in the *dsl.component* decorator. We would need kubeflow-pytorchjob, kubeflow-training[huggingface] and numpy packages in this Kubeflow component.\n",
+    "\n",
+    "Replace the HUGGINGFACE_TOKEN with your own token. It should have access to [Llama3 model](https://huggingface.co/meta-llama/Llama-3.1-8B-Instruct)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2a4d1541-0892-490b-b9b7-fc4057cce174",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@dsl.component(packages_to_install=['kubeflow-pytorchjob', 'kubeflow-training[huggingface]','numpy<1.24'])\n",
```

Comment:
Dito, is `kubeflow-pytorchjob` really necessary since `TrainingClient` is used now?